### DISCUSSION TITLE
Multi-part forming losing data on step change

### DISCUSSION THREAD

**Discussion by AleKrabbe** [react-hook-form/react-hook-form#2414] [Q&A] [answered]
Hello, 

I'm trying to implement a multipart form using the `<Stepper />` from Material-UI, I've found @bluebill1049 [video](https://www.youtube.com/watch?v=CeAkxVwsyMU) on how to do this using `little-state-machine` and using one form per step. However; in my case, I would like to manually use `redux` and `react-redux`, the concept seemed simple enough... Save your fields data on state and use that state as the default value on `useForm()`. This is not working for me, state does update, but when moving from one step to the other, the fields in that step are reset, also I only use a single form that encompasses all the fields in each step. Is anyone able to help me identify what going on here? I made a simple CSB to workout this issue. Thanks!

I added console messages to help figure out what is going on in the state. To move between steps, just click the step number on the top of the page.

[![Edit ecstatic-northcutt-uxtee](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/ecstatic-northcutt-uxtee?fontsize=14&hidenavigation=1&theme=dark)

**Comment by bluebill1049** [comment, marked as answer]
take a look `shouldUnregister` option https://react-hook-form.com/api#useForm

**Reply by AleKrabbe** [reply]
Thank you, that solved my problem.
